src/crab/var_factory.cpp

```diff
@@ -35,6 +34,13 @@ void variable_t::clear_thread_local_state() {
         "r10.value", "r10.ctx_offset", "r10.map_fd", "r10.packet_offset", "r10.shared_offset", "r10.stack_offset", "r10.type", "r10.shared_region_size", "r10.stack_numeric_size",
         "data_size", "meta_size",
     };
+};
+
+thread_local crab::lazy_allocator<std::vector<std::string>, variable_t::variable_name_factory> variable_t::names;
+
+void
+variable_t::clear_thread_local_state() {
+    names.clear();
```

Comment:
This state is immutable. Why and when does it need to be cleared? I think it's only thread-local to guarantee no race conditions during initialization (which can also be taken care of by putting it in a function-static variable, though even for  a simple global variable it's unlikely to have an issue).

Reply:
Also, if I understand correctly, this is the only variable that requires factory as a template parameter.